### DISCUSSION TITLE
"Artifact update problem" should override-disable auto-merge

### DISCUSSION THREAD

**Discussion by TWiStErRob** [renovatebot/renovate#30067] [Suggest an Idea]
### Tell us more.

In this case, it doesn't really matter how the "Artifact update problem" came about, but based on the wording vs the actions, Renovate gives no recourse to humans to review the problem.

<details><summary>See the timeline of TWiStErRob/net.twisterrob.colorfilters#335</summary>

![image](https://github.com/renovatebot/renovate/assets/2906988/f10e5b41-44c6-41c5-9dd4-aa04df9bbb00)

</details>

---

Note: there's precedence for this.

![image](https://github.com/renovatebot/renovate/assets/2906988/844b6ac7-56e5-4bec-aa32-e0c21a0effe3)

**Comment by rarkins** [comment]
Important distinction - this is specifically about platformAutomerge where github makes the decision to merge.

**Reply by TWiStErRob** [reply]
No, Renovate makes a decision to "merge when ready":

![image](https://github.com/renovatebot/renovate/assets/2906988/f57fe617-b0bd-422e-91e9-a3c64daec6be)

then CI executes, and upon completion GitHub does the actual merge of the PR.

Renovate could also disable auto-merge as soon as it realizes that things are broken (which is actually **BEFORE** (platform) auto-merge is enabled). In human terms, one wouldn't press the (auto-)merge button when they know that someone needs to look at the PR, regardless of whether platform or non-platform strategy is used.

**Reply by rarkins** [reply]
I agree with everything you wrote except the initial "No". GitHub makes the decision whether or not to merge, but like you wrote - Renovate could decide not to enable it at all if it knows it has a failing check.

**Reply by TWiStErRob** [reply]
Yes, that's what I was aiming for with this report.

(The "No" was for the **decision to start** the merging process, which eventually ends up in a commit on base branch ("merge"). I guess it's just a different mental model we have about the complexities of merging.)

Based on the precedence image in OP, there's already facilities to achieve this, so it could be an easy implementation?

**Reply by rarkins** [reply]
Yes it's probably reasonably easy.  Renovate now has logic to restart platform automerge after new commits so that needs to be updated too.

**Comment by rarkins** [comment]
Converted to issue: https://github.com/renovatebot/renovate/issues/30269
